clients/stellar-relay/src/connection/services.rs

```diff
@@ -277,6 +279,12 @@ pub(crate) async fn connection_handler(
 	loop {
 		match timeout(Duration::from_secs(connector.timeout_in_secs), actions_receiver.recv()).await
 		{
+			Ok(Some(ConnectorActions::Disconnect)) => {
+				log::error!("dropped");
+				drop(w_stream);
+				return Ok(())
+			},
+
```

Comment:
Is there a reason why you put this here and not into the `_connection_handler()` where the other actions are handled?

Reply:
There reason why i put this here it is because that inside of _connection_handler we have unique reference
```
async fn _connection_handler(
	actions: ConnectorActions,
	connector: &mut Connector,
	w_stream: &mut tcp::OwnedWriteHalf,
) -> Result<(), Error> {...}
```

We need to call drop(..) function for value and not for reference.


```
#[derive(Debug)]
struct X(i32);
impl Drop for X{
    fn drop(&mut self){
        println!("DROP");
    }
}

fn main() {
    let mut x = X(1);
    r(&mut x);
    println!("{:#?}", x);
    drop(x);
    
}

fn r(v : &mut X)
{
    println!("try drop");
    drop(v);
    println!("try drop");
}

```